### ISSUE TITLE
could you please add pub/sub support? 

### ISSUE THREAD

**Issue by sddongxh** [Nordix/hiredis-cluster#27]
Thanks.

**Comment by andreasgerstmayr** [comment]
Reading https://redis.io/commands/pubsub you can send `PUBLISH`/`SUBSCRIBE` to any node in the cluster. You can use `redisClusterAsyncCommandToNode()` to send a command directly to a specific node.

Should hiredis-cluster send the command a specific node by hashing the channel (as it does for regular Redis keys)?

**Comment by zuiderkwast** [comment]
> Should hiredis-cluster send the command a specific node by hashing the channel (as it does for regular Redis keys)?

No, pubsub channels are not sharded like keys. They are forwarded between the Redis nodes in a cluster, so you can subscribe to any node and publish to any node. (Pubsub doesn't scale that well in a large cluster since all messages are forwarded to all nodes, but that's different topic.)

I'm not sure what is the best way to implement this in hiredis-cluster but it can be good to know that when using pubsub, you typically need a dedicated connection to receive published messages. The documentation of SUBSCRIBE says:

> Once the client enters the subscribed state it is not supposed to issue any other commands, except for additional SUBSCRIBE, PSUBSCRIBE, UNSUBSCRIBE, PUNSUBSCRIBE, PING, RESET and QUIT commands.
